tests/rptest/tests/compatibility/sarama_scram_test.py

```diff
@@ -0,0 +1,101 @@
+class SaramaScramTest(RedpandaTest):
+
+    CLIENT_DIR = "/kafka-clients"
+    TESTS_DIR = os.path.join(CLIENT_DIR, "sarama")
+
+    def __init__(self, test_context):
+        extra_rp_conf = dict(
+            developer_mode=True,
+            enable_sasl=True,
+        )
+        super(SaramaScramTest, self).__init__(test_context,
+                                        extra_rp_conf=extra_rp_conf)
+
+        self.sarama_topic = "sarsecure"
+
+        # Check if CLIENT Dir exists first
+        try:
+            os.mkdir(SaramaScramTest.CLIENT_DIR)
+        except FileExistsError:
+            self.logger.debug("Not creating {}. It already exists".format(SaramaScramTest.CLIENT_DIR))
+
+        # Check if sarama dir already exists first
+        if os.path.isdir(SaramaScramTest.TESTS_DIR):
+            self.logger.debug("Not downloading {}. It already exists".format(SaramaScramTest.TESTS_DIR))
+        else:
+            self._execute("git clone https://github.com/Shopify/sarama.git", SaramaScramTest.CLIENT_DIR)
+
+    def _execute(self, cmd, folder=None):
+        self.logger.debug("Executing command: %s", cmd)
+        try:
+            res = subprocess.check_output(cmd,
+                                          stderr=subprocess.STDOUT,
+                                          shell=True,
+                                          cwd=folder)
+            self.logger.debug(res)
+            return res
+        except subprocess.CalledProcessError as e:
+            self.logger.debug("Error (%d) executing command: %s",
+                                        e.returncode, e.output)
+
+    def make_superuser_client(self, password_override=None):
+        username, password, algorithm = self.redpanda.SUPERUSER_CREDENTIALS
+        password = password_override or password
+        return PythonLibrdkafka(self.redpanda,
+                                username=username,
+                                password=password,
+                                algorithm=algorithm), (username, password)
+
+    @cluster(num_nodes=3)
+    def test_sarama_sasl_scram(self):
+        EXAMPLE_DIR = os.path.join(SaramaScramTest.TESTS_DIR, "examples/sasl_scram_client")
+
+        #Tidy the example
+        self._execute("go mod tidy", EXAMPLE_DIR)
+
+        #Build the example
+        self._execute("go build", EXAMPLE_DIR)
+
+        # #Create ACL user and topic authenticated to the user
+        client, creds = self.make_superuser_client()
+        client.create_topic(TopicSpec(name=self.sarama_topic))
+        topics = client.topics()
+        print(topics)
+
+        #Run the example
+        cmd = "./sasl_scram_client -brokers n1:9092 -username {0} -passwd {1} -topic {2} -algorithm sha256".format(creds[0], creds[1], self.sarama_topic)
+        p = subprocess.Popen([cmd],
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.STDOUT,
+                        shell=True,
+                        cwd=EXAMPLE_DIR)
```

Comment:
once the sasl client tool is build in the Dockerfile it will be available on all the nodes that are part of the ducktape cluster. In this case you can then avoid invoking the command locally, and instead use something like `node.ssh_command(nodes[0], cmd)` (its built-in to ducktape but that is from memory--check out some of the other services for examples), and ducktape will automatically collect the output etc for you so you don't need to do that here.

Reply:
That's pretty cool. I will make the change.